src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -152,6 +152,16 @@ public UUID createAndLock(Snapshot snapshot, String flightId) {
         return snapshotId;
     }
 
+    protected String getExclusiveLock(UUID id) {
```

Comment:
The method name sounds like it is actually taking out the lock. Maybe `getExclusiveLockState`